lib/fuse_lowlevel.c

```diff
@@ -2940,7 +2940,7 @@ int fuse_req_getgroups(fuse_req_t req, int size, gid_t list[])
 		goto out_free;
 	}
 
-	if (ret == bufsize) {
+	if ((size_t)ret == bufsize) {
```

Comment:
I think to make this safe, the condition above has to be changed from `ret == -1` to `ret < 0`.